src/Umbraco.Core/PropertyEditors/ValueConverters/SliderValueConverter.cs

```diff
@@ -11,77 +8,73 @@ namespace Umbraco.Core.PropertyEditors.ValueConverters
     [DefaultPropertyValueConverter]
     public class SliderValueConverter : PropertyValueConverterBase
     {
-        private readonly IDataTypeService _dataTypeService;
+        public SliderValueConverter()
+        { }
 
+        [Obsolete("This constructor isn't required anymore, because we don't need services to lookup the data type configuration.")]
         public SliderValueConverter(IDataTypeService dataTypeService)
-        {
-            _dataTypeService = dataTypeService ?? throw new ArgumentNullException(nameof(dataTypeService));
-        }
+        { }
 
         public override bool IsConverter(IPublishedPropertyType propertyType)
             => propertyType.EditorAlias.InvariantEquals(Constants.PropertyEditors.Aliases.Slider);
 
         public override Type GetPropertyValueType(IPublishedPropertyType propertyType)
-            => IsRangeDataType(propertyType.DataType.Id) ? typeof (Range<decimal>) : typeof (decimal);
+            => IsRangeDataType(propertyType) ? typeof(Range<decimal>) : typeof(decimal);
 
         public override PropertyCacheLevel GetPropertyCacheLevel(IPublishedPropertyType propertyType)
             => PropertyCacheLevel.Element;
 
         public override object ConvertIntermediateToObject(IPublishedElement owner, IPublishedPropertyType propertyType, PropertyCacheLevel cacheLevel, object source, bool preview)
         {
-            if (source == null)
+            var sourceString = source?.ToString();
+            if (string.IsNullOrEmpty(sourceString))
+            {
                 return null;
+            }
 
-            if (IsRangeDataType(propertyType.DataType.Id))
+            if (IsRangeDataType(propertyType))
             {
-                var rangeRawValues = source.ToString().Split(',');
+                // Return range
+                var rangeRawValues = sourceString.Split(',');
                 var minimumAttempt = rangeRawValues[0].TryConvertTo<decimal>();
-                var maximumAttempt = rangeRawValues[1].TryConvertTo<decimal>();
 
-                if (minimumAttempt.Success && maximumAttempt.Success)
+                if (rangeRawValues.Length == 1 && minimumAttempt.Success)
+                {
+                    // Configuration is probably changed from single to range, return range with same min/max
+                    return new Range<decimal>
+                    {
+                        Minimum = minimumAttempt.Result,
+                        Maximum = minimumAttempt.Result
+                    };
+                }
```

Comment:
When changing the configuration to enable ranges, the saved data is still a single value, so let's use that as both minimum and maximum value.